python/lsst/daf/butler/formatters/parquet.py

```diff
@@ -111,21 +114,36 @@ def _standardizeColumnParameter(self, columns: Dict[str, Union[str, List[str]]])
         name : `str`
             Stringified tuple representing a multi-index column name.
         """
-        if not isinstance(columns, collections.abc.Mapping):
-            raise ValueError("columns parameter for multi-index data frame must be a dictionary.")
-        if not set(self.indexLevelNames).issuperset(columns.keys()):
-            raise ValueError(f"Cannot use dict with keys {set(columns.keys())} "
-                             f"to select columns from {self.indexLevelNames}.")
-        factors = [iterable(columns.get(level, self.columns.levels[i]))
-                   for i, level in enumerate(self.indexLevelNames)]
-        for requested in itertools.product(*factors):
-            for i, value in enumerate(requested):
-                if value not in self.columns.levels[i]:
-                    raise ValueError(f"Unrecognized value {value!r} for index {self.indexLevelNames[i]!r}.")
-            yield str(requested)
-
-    def read(self, columns: Union[str, List[str], Dict[str, Union[str, List[str]]]] = None
-             ) -> pd.DataFrame:
+        if isinstance(columns, list):
+            for requested in columns:
+                if not isinstance(requested, tuple):
+                    raise ValueError(
+                        "columns parameter for multi-index data frame must be either a dictionary or list of tuples."
+                    )
+                yield str(requested)
```

Comment:
It seems odd to have made sure it is a list of tuples to immediately force it to be a string. Why not make it a string initially?

Reply:
The reason for all this gymnastics about the column names is because of funniness about how pyarrow deals with the column names of a table with a multi-level column index.  

In pandas alone, if you have a multilevel column index, you can index it with a straight-up tuple; e.g. `df[('meas', 'HSC-G', 'coord_ra')]` to grab that specific column.  To read that same column from a parquet table via pyarrow, however, you would have to pass `columns=["('meas', 'HSC-G', 'coord_ra')"]` to the parquet loader.  This is tiresome to have to remember to do in the functor front-end, and there are situations where requesting columns via the "pass a dictionary of column specifications which gets converted internally to the list of columns needed from pyarrow" doesn't work, so we need to pass the list of tuples (e.g., for composite functors, color functors, and maybe more).  So rather than needing to string-ify them within the `Functor` implementation or call, we do the string-ifying in the butler parquet loader.  This way also, the same `Functor` code could in principle be used directly on a dataframe, not necessarily on the dataref that will load from parquet.